ansible_builder/main.py

```diff
@@ -44,13 +47,15 @@ def build_command(self):
         ]
 
     def build(self):
+        self.containerfile.prepare_prepended_steps()
         self.containerfile.prepare_introspection_steps()
         self.containerfile.prepare_galaxy_steps()
-        print('Writing partial Containerfile without collection requirements')
+        print(MessageColors.OK + 'Writing partial Containerfile without collection requirements' + MessageColors.ENDC)
         self.containerfile.write()
         run_command(self.build_command)
         self.containerfile.prepare_pip_steps()
-        print('Rewriting Containerfile to capture collection requirements')
+        print(MessageColors.OK + 'Rewriting Containerfile to capture collection requirements' + MessageColors.ENDC)
```

Comment:
This is really nicely written.

If I was going to point to the next thing to do, I would prefer to route this through an internal `debug` method, and pass the value OK/FAIL/etc. to that method, just so that the main method isn't touching the colors directly. But that's thinking about additional enhancement, and what you have is 👍